source/develop/release-management/features/network/lldp.html.md

```diff
@@ -0,0 +1,277 @@
+  * System Description TLV (Type = 6)
+  * Management Address TLV (Type = 8)
+  has to be available on the engine's REST API
+
+* Is there an upper limit of the acceptable age of the data?
+
+## Entity Description
+
+The initial LLDP information is a list of type, length, value (TLV) tuples.
+To express multiple values, e.g. to list the allowed VLANs on a switch port,
+the TLV is sent multiple times.
+The value may contain structured data.
+This way the LLDP data is non-uniform.
+This non-uniform LLDP data will be mapped to the following uniform entities.
+
+### tlv
+
+A `tlv` is associated to a `vds_interface`, has a `name`, a `description` and a
+`value`.
+
+The `description` is a human readable string to describe what the value is about.
+
+The `name` is a string created by rules, to allow a bidirectional mapping
+between the name and a tuple of `type`, `oui` and `subtype` of the LLDP's
+original TLV. This would result in names similar the ones used in OpenStack [4].
+
+The `value` may be a single simple string, or an ordered list of `tlv_value`s.
+
+There will be a new table in engine to store the `tlv`s. The table needs columns
+for `id`, `vds_interface_id`, `name`, `description` and `value`.
+
+### tlv_value
+A `tlv_value` is a single list entry of the ordered list of `tlv`'s `tlv_value`s.
+
+There will be a new table in engine to store the `tlv_value`s. The table needs
+columns for `id`, `tlv_id`, `description`, `value`, `position`.
+
+
+## CRUD
+
+TODO: Describe the create/read/update/delete operations on the entities, and what each operation should perform.
+
+
+## Gathering the LLDP data from the hosts
+
+There are multiple alternatives for engine to gather the LLDP data from the hosts:
+
+* Extend VDSM to provide the LLDP data
+
+* Use ansible's lldp module [3]
+
+Because the Ansible module is flagged as preview and requires lldpd, which is
+in only in EPEL repository for RHEL, VDSM has to be extended.
+
+## VDSM implementation options
+
+to be imported from https://github.com/EdDev/ovirt-devwiki/blob/master/design/network/lldp/lldp.md
+
+## Communication of VDSM and engine
+
+TODO
+
+Are incomplete TLV values allowed?
+
+Timestamp by VDSM or engine?
+
+## User Experience
+
+### REST-API
+
+The lldp information are read-only. They could be read by a request like
+
+```
+GET https://ovirtexampleengine/ovirt-engine/api/hosts/123/nics/321/lldap
+```
+
+The response in XML will look like this
+
+```
+<lldap>
+  <tlv>
+    <!-- example for simple string -->
+    <name>chassis_id</name>
+    <descripton>Chassis ID</desciption>
+    <value>64:64:9b:xx:xx:xx</value>
+  </tlv>
+  <tlv>
+    <!-- example for simple string -->
+    <name>port_description</name>
+    <descripton>Port Description</desciption>
+    <value>This is a string longer than 1000 chars and contains / </value>
+  </tlv>
+  <tlv>
+    <!-- example for structured data -->
+    <name>management_address</name>
+    <descripton>Management Address</desciption>
+    <value>
```

Comment:
Why do we need two levels for this? Can't we just have multiple value elements?
ID/s are needed here as well, to make it easy for the user (that is not a human) to understand what each value is.

I'm also not so clear what a single "structured" value represents, is it defined (by the standards) as an inner TLV? If so, then having the proposed two levels here makes sense, but it is not a tlv_value, just a tlv or subtlv.